x-pack/plugin/runtime-fields/qa/core-with-search/src/yamlRestTest/java/org/elasticsearch/xpack/runtimefields/test/search/CoreTestsWithSearchRuntimeFieldsIT.java

```diff
@@ -0,0 +1,230 @@
+                    }
+                    return true;
+                }
+
+                private Map<?, ?> runtimeMappings(String index) {
+                    if (index == null) {
+                        return mergeMappings(new String[] { "*" });
+                    }
+                    String[] patterns = Arrays.stream(index.split(",")).map(m -> m.equals("_all") ? "*" : m).toArray(String[]::new);
+                    if (patterns.length == 0 && Regex.isSimpleMatchPattern(patterns[0])) {
+                        return runtimeMappings.get(patterns[0]);
+                    }
+                    return mergeMappings(patterns);
+                }
+
+                private Map<?, ?> mergeMappings(String[] patterns) {
+                    Map<String, Map<String, Object>> merged = new HashMap<>();
+                    for (Map.Entry<String, Map<String, Map<String, Object>>> indexEntry : runtimeMappings.entrySet()) {
+                        if (false == Regex.simpleMatch(patterns, indexEntry.getKey())) {
+                            continue;
+                        }
+                        for (Map.Entry<String, Map<String, Object>> field : indexEntry.getValue().entrySet()) {
+                            Map<String, Object> mergedConfig = merged.get(field.getKey());
+                            if (mergedConfig == null) {
+                                merged.put(field.getKey(), field.getValue());
+                            } else if (false == mergedConfig.equals(field.getValue())) {
+                                // The two indices have different runtime mappings for a field so we have to give up on running the test.
+                                return null;
+                            }
+                        }
+                    }
+                    for (Map.Entry<String, Set<String>> indexEntry : mappedFields.entrySet()) {
+                        if (false == Regex.simpleMatch(patterns, indexEntry.getKey())) {
+                            continue;
+                        }
+                        for (String mappedField : indexEntry.getValue()) {
+                            if (merged.containsKey(mappedField)) {
+                                // We have a runtime mappings for a field *and* regular mapping. We can't make this test work so skip it.
+                                return null;
+                            }
+                        }
+                    }
+                    return merged;
+                }
+
+                @Override
+                protected boolean handleIndex(IndexRequest index) {
+                    /*
+                     * Ok! Let's reverse engineer dynamic mapping. Sort of. We're
+                     * really just looking to figure out which of the runtime fields
+                     * is "close enough" to what dynamic mapping would do.
+                     */
+                    if (index.getPipeline() != null) {
```

Comment:
I guess at some point we're going to need to integrate runtime fields into dynamic mappings in core code?

Reply:
The "mapped" version of these tests gets to use dynamic mappings. The "search request" version has to use dynamic mappings to disable the fields so that we can "prove" that we're replacing them with search requests.

I guess another option is for me to hack the search requests and issue a get mapping or field caps request before hand to try to construct the search request based on the disabled mappings that I made. I didn't have that option when I first wrote the code so I did the dynamic mapping stuff. Its kind of nasty so maybe I should replace it with that...... I'd be ok doing it now if you feel strongly about it but I'd also be ok leaving this for a while.

Reply:
I think I'm getting confused between dynamic mappings and dynamic templates anyway.  Let's leave it as it is, but maybe open a separate issue to consider reworking it so it doesn't get lost?

Reply:
#64716